src/core/jpg.js

```diff
@@ -824,6 +824,12 @@ var JpegImage = (function JpegImageClosure() {
               // block was eaten by the encoder
               offset -= 3;
               break;
+            } else if (data[offset] === 0xFF &&
+                       data[offset + 1] >= 0xC0 && data[offset + 1] <= 0xFE) {
+              // If the length of the previous block is incorrect (too short),
+              // treat the current "marker" as padding and skip past it when
+              // the next marker is a valid one (fixes issue8169.pdf).
+              break;
```

Comment:
Good point; is the updated patch better?
*Note:* I've not had time to run all tests locally yet, so if the approach is OK I'll trigger testing later.